lib/sql.js

```diff
@@ -912,6 +971,42 @@ SQLConnector.prototype.buildFields = function(model, data, excludeIds) {
 };
 
 /**
+ * Build an array of fields for the replace database operation
+ * @param {String} model Model name
+ * @param {Object} data Model data object
+ * @param {Boolean} excludeIds Exclude id properties or not, default to false
+ * @returns {{names: Array, values: Array, properties: Array}}
+ */
+SQLConnector.prototype.buildReplaceFields = function(model, data, excludeIds) {
```

Comment:
I believe the same comment applies here as well: "buildReplaceFieldsis a protected method subject to override by subclasses."
Please see the PR from `loopback-connector-mysql`; it is being used there...